README.md

```diff
@@ -41,54 +44,132 @@ Upon successful build the binary will be available in `build/ib-kubernetes`.
+#### Building image mellanox/ib-kubernetes
+```
 $ make image
+```
 
-# Building image with custom tag and Dockerfile
+#### Building image with custom tag and Dockerfile
+```
 $ DOCKERFILE=myfile TAG=mytag make image
 ```
 
 ## Configuration Reference
 
-User can provide the following configurations as environment variables or for the ConfigMap :
-* `DAEMON_SM_PLUGIN`: Name of the subnet manager plugin. Currently supported `"noop"` and `"ufm"`.
-* `DAEMON_PERIODIC_UPDATE`: Interval in seconds to send add and remove request to subnet manager.
-* `GUID_POOL_RANGE_START`: The first guid in the pool e.g: `"02:00:00:00:00:00:00:00"`.
-* `GUID_POOL_RANGE_END`: The Last guid in the pool.
+IB Kubernetes configration as ConfigMap :
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ib-kubernetes-config
+  namespace: kube-system
+data:
+  DAEMON_SM_PLUGIN: "ufm" # Name of the subnet manager plugin
+  DAEMON_PERIODIC_UPDATE: "5" # Interval in seconds to send add and remove request to subnet manager
+  GUID_POOL_RANGE_START: "02:00:00:00:00:00:00:00" # The first guid in the pool
+  GUID_POOL_RANGE_END: "02:FF:FF:FF:FF:FF:FF:FF" # The last guid in the pool
+```
+
+## Plugins
+
+Subnet Manager Plugin to configure PKeys (Partition Keys) in the InfiniBand fabric.
+
+### NOOP Plugin
+
+Plugin that does nothing. Example for developing user subnet manager plugin
+
+### UFM (Unified Fabric Manager) Plugin
+
+[UFM](https://www.mellanox.com/products/management-software/ufm) is a powerful platform for managing scale-out computing environments.
+UFM Plugin allow to configure PKeys (Partition Keys) via UFM.
+
+#### Plugin Configuration
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ib-kubernetes-ufm-secret
+  namespace: kube-system
+stringData:
+  UFM_USERNAME: "admin"  # UFM Username
+  UFM_PASSWORD: "123456" # UFM Password
+  UFM_ADDRESS: ""        # UFM Hostname/IP Address 
+  UFM_HTTP_SCHEMA: ""    # http/https. Default: https
+  UFM_PORT: ""           # UFM REST API port. Defaults: 443(https), 80(http)
+string:
+  UFM_CERTIFICATE: ""    # UFM Certificate in base64 format. (if not provided client will not verify server's certificate chain and host name)
+```
+
+#### Self Signed Certificates
 
-**Configurations if "ufm" subnet manager plugin is used for  `deployment/ib-kubernetes-ufm-secret.yaml`:**
-* `UFM_USERNAME`: Username of UFM.
-* `UFM_PASSWORD`: Password of UFM.
-* `UFM_ADDRESS`: IP address or hostname of UFM server.
-* `UFM_HTTP_SCHEMA`: http/https, default is https.
-* `UFM_PORT`: REST API port of UFM default is 443 (https), if `httpSchema` is set to http then the default is 80.
-* `UFM_CERTIFICATE`: Secure certificate if using secure connection.
+##### Login to UFM
+
+Containerized UFM:
+``` 
+$ docker exec -it ufm bash
+```
+
+##### Create private key and certificate
+```
+$ openssl req -x509 -newkey rsa:4096 -keyout ufm.key -out ufm.crt -days 365 -subj '/CN=<UFM hostname>'
+```
+
+##### Install UFM private key and certificate
```

Comment:
the installation of certificate is regardless to when self-signed are used or provided by the user right ?
so perhaps move the headline back i.e 4x#

Reply:
I restructure it, let me know what you think